### PR TITLE
NAS-123677 / 24.10 / Improve error handling for already in use PCI devices

### PR DESCRIPTION
**Pull request by Qubad786**
**Problem:**

An issue arises when the same PCI device is added to two virtual machines (VMs), and one of the VMs is already running. When the user attempts to start the second VM, QEMU raises an error indicating that the PCI device is already in use by another VM.

**Solution:**

To address this problem, validation for the PCI device has been updated. It now also considers whether the same PCI device is in use by another VM. If the PCI device is not in use by any other VM, it is marked as available for use or otherwise error out on middleware side that the device is already being used.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/vm/devices/pci.py [hunk 1]
```diff
@@ -37,6 +37,10 @@ def reattach_device(self):
 
     def pre_start_vm_device_setup(self, *args, **kwargs):
         device = self.get_details()
+
+        if not self.not_in_use_by_vm():
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/vm/devices/pci.py:41]: This is a double-negative which makes it pretty hard to follow what's actually being checked. I also think the name `not_in_use_by_vm()` is actually the inverse of what it's checking? It's actually checking if the device is IN USE by a vm? Either way, this is very confusing so we should rename/fix it.

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Comment by Migsi** [comment]
Wouldn't it be better to just highlight in use devices instead of removing them from the list alltogether? I'm running a setup where I pass through the same GPU to a Windows and a Linux VM, which are running mutually exclusive. While obviously not done every day, having to stop either VM before being able to setup (not run) a second, thrid, fourth, etc. still seems quite a bit annoying to me.

**Comment by Qubad786** [comment]
@Migsi yes that is accurate so your case is still good to go ✅

**Review by yocalebo — APPROVED** [review]
